xarray/core/rolling.py

```diff
@@ -351,6 +352,14 @@ def _bottleneck_reduce(self, func, **kwargs):
     def _numpy_or_bottleneck_reduce(
         self, array_agg_func, bottleneck_move_func, **kwargs
     ):
+        if "dim" in kwargs:
+            warnings.warn(
```

Comment:
Should this just be a `UserWarning` saying that the `dim` kwargs is ignored?

Reply:
If it's definitely incorrect to pass (i.e. there's no legitimate potential use), I'd vote to raise an error in the future, rather than a warning. If someone makes a mistake, we should let them know quickly & loudly.